### ISSUE TITLE
Suggestion: Make an Exception to `\newcommand` When `--mathjax` is Used

### ISSUE THREAD

**Issue by ickc** [jgm/pandoc#2762]
i originally put this in #2758 since they are closely related. But since it is really a different issue and requires different solution, I opened it here instead:

# Issue: MathJax Requires Extra Math Delimiters When `\newcommand` is Used

Because of [MathJax TeX and LaTeX Support — MathJax 2.6 documentation](https://mathjax.readthedocs.org/en/latest/tex.html#defining-tex-macros), MathJax expects an extra delimiter on `\newcommand` and the like. But if that extra delimiter is used, HTML+MathJax would be fine, but LaTeX is not. See example in [mathjax-latex-md-mmd-pandoc/M!WE-pandoc.md at gh-pages · ickc/mathjax-latex-md-mmd-pandoc](https://github.com/ickc/mathjax-latex-md-mmd-pandoc/blob/gh-pages/newcommand/M!WE-pandoc.md).

# The Problems of the Current Behavior of pandoc

If used in pure pandoc with no extra delimiter, like in [mathjax-latex-md-mmd-pandoc/M!WE-pandoc2.md at gh-pages · ickc/mathjax-latex-md-mmd-pandoc](https://github.com/ickc/mathjax-latex-md-mmd-pandoc/blob/gh-pages/newcommand/M!WE-pandoc2.md), pandoc recognized the macro and would replace all `\newcommand` I defined with the corresponding code. But then since pandoc don't recognize `\begin{align}...\end{align}` and the like as math and process it (related to #2758), the result would also be incorrect.

Moreover, it could be argued that the `\newcommand` should leave as is and let MathJax to process them to minify the resulting HTML and also make the generated HTML closer to the source and more readable.

# Suggestion

My suggestion is, if `--mathjax` is used, rather than processing the `\newcommand`, an extra delimiter like `\(...\)` should be put around it, and otherwise leaves as verbatim.

And lastly, similar to the above suggestion, `\ renewcommand` should be treated similarly.

# Summary

In short, I suggest that in addition to

> In LaTeX output, the `\newcommand` definition will simply be passed unchanged to the output.

we add the following behavior:

> In HTML output with `--mathjax` option, a pair of math delimiters `\(...\)` will be used to enclose the `\newcommand` definition, and otherwise simply be passed unchanged to the output.

The inclusion of `\ renewcommand` is implicit.

**Comment by ickc** [comment]
Oops! I just found out it is a duplicate of #1393. Sorry!
